hydroshare/settings.py

```diff
@@ -504,7 +508,7 @@ def _(s): return s # noqa
     "PAGE_SIZE_QUERY_PARAM": "PAGE_SIZE",
     "EXCEPTION_HANDLER": "rest_framework.views.exception_handler",
     "DEFAULT_AUTHENTICATION_CLASSES": (
-        "rest_framework.authentication.BasicAuthentication",
```

Comment:
@devincowan @pkdash  - Can you help me think through the consequences of this?  I think most users of our API use hsclient so we should be able to update hsclient to use the updated authentication.  I'm not sure what to do about applications that are using basic authentication directly.

We could leave basic authentication in and as long as the user is not migrated to keycloak then there is no issue.  It seems reasonable that a user used within an application won't be migrated to keycloak.

Reply:
Feels reasonable to me to leave BasicAuth in. What would happen if a user that has migrated to Keycloak attempted to BasicAuth to the api? Would it provide any substantive error message by way of explanation?

I think if we are going to remove basic auth altogether, we at the very least have to provide a deprecation warning for x months before doing so -- so we would us a [sunset header](https://datatracker.ietf.org/doc/html/rfc8594). I think we could/should alter our metrics if needed so we can detect who is using the api with basic auth _not through hsappi_. We monitor it for say 6mo while we have the sunset header, collecting account info and notifying by email those users who are using basic auth for hsapi. Then we make the cutover and remove the basic auth functionality.

We also would document the endpoints as deprecated within our swagger docs (looks like that's possible in v3).

Another option that seems worse would be to version our api and keep the current v1 open for basic auth. It's just a brainstorm, seems like major drawbacks without any benefit other than consistent user experience compared with historical precedent.

Reply:
>Feels reasonable to me to leave BasicAuth in. What would happen if a user that has migrated to Keycloak attempted to BasicAuth to the api? Would it provide any substantive error message by way of explanation?

I think it would just work until the password changes.  If the user changed the password in keycloak it would not change in django.  The user will not be able to change their password in django.

Reply:
It seems from the discussion BasicAuth and keycloak can co-exist. In that case we should keep BasicAuth as long as we can. I believe SWATShare is using BasicAuth. I assume at some point we have to stop supporting basicAuth.

Reply:
Yeah, I think we should implement the sunset header, document the change and plan to support both for X months before turning of basic auth entirely.

Reply:
Makes sense to me!

Reply:
I deployed the change with basic auth back in and tested that it works as described in this thread.  I'll get started on updating hsclient to support the oidc authentication.